stdlib/mlang/language-composer.mc

```diff
@@ -0,0 +1,355 @@
+include "ast.mc"
+include "ast-builder.mc"
+
+include "mexpr/info.mc"
+
+include "bool.mc"
+include "common.mc"
+include "error.mc"
+include "tuple.mc"
+include "name.mc"
+include "seq.mc"
+include "map.mc"
+
+type DeclInfo
```

Comment:
How do these relate to the `Decl` type? Are they the same as the corresponding constructors, or do they carry different data?

Reply:
They carry a subset of the data. I thought it would be rather expensive to carry all of the cases and constructors around so I created this `Info` type.

Reply:
Probably document the connection.

(Depending on precise needs it might not be an issue to carry the original `Decl` around, perhaps with a subsetted type, since it's already a value that exists, while this will have to create a new value with a subset of the data. Probably reasonable to have a separate type though if the extra data isn't needed, and especially if it would get in the way somehow.)